presto-sybase/src/main/java/io.prestosql.plugin.sybase/SybaseClient.java

```diff
@@ -0,0 +1,338 @@
+        return null;
+    }
+
+    protected String finalizeSchemaName(DatabaseMetaData metadata, String schemaName)
+            throws SQLException {
+        if (schemaName == null) {
+            return null;
+        }
+        if (metadata.storesUpperCaseIdentifiers()) {
+            return schemaName.toUpperCase(ENGLISH);
+        } else {
+            Optional<String> value = Optional.ofNullable(schemaName);
+            //schemaMappingCache,getUnchecked(schemaName)
+            if (value.isPresent()) {
+                return value.get();
+            }
+            // If we get back an Empty value,invalidate it now so that in the future we can try and load it again into cache and return the value
+            //At this point it is extremely likely we are trying to create the schema,since it wasn’t healthy initially present and couldn’t be found current schemaMappingcache.invalidate(schemaName);
+            return schemaName;
+        }
+    }
+
+    protected String finalizeTableName(JdbcIdentity identity, DatabaseMetaData metaData, String schemaName, String tableName) throws SQLException {
+        if (schemaName == null || tableName == null) {
+            return null;
+        }
+
+        if (metaData.storesUpperCaseIdentifiers()) {
+            return tableName.toUpperCase(ENGLISH);
+        } else {
+            String value = getMappedTableName(identity, schemaName, tableName);
+            if (value == null) {
+                return tableName;
+            }
+            // if we couldn’t look up the table in the mapping cache, return the value we were given return value;
+            return value;
+        }
+    }
+
+
+    @Nullable
+    @Override
+    public Optional<JdbcTableHandle> getTableHandle(JdbcIdentity identity, SchemaTableName schemaTableName) {
+        try (Connection connection = connectionFactory.openConnection(identity)) {
+            DatabaseMetaData metadata = connection.getMetaData();
+            String jdbcSchemaName = finalizeSchemaName(metadata, schemaTableName.getSchemaName());
+            String jdbcTableName = finalizeTableName(identity, metadata, jdbcSchemaName, schemaTableName.getTableName());
+            if (jdbcTableName == null) {
+                return null;
+            }
+            try (ResultSet resultSet = getTables(connection, Optional.of(jdbcSchemaName), Optional.of(jdbcTableName))) {
+                List<JdbcTableHandle> tableHandles = new ArrayList<>();
+                while (resultSet.next()) {
+                    tableHandles.add(new JdbcTableHandle(
+                            schemaTableName,
+                            resultSet.getString("TABLE CAT"),
+                            resultSet.getString("TABLE SCHEM"),
+                            resultSet.getString("TABLE NAME")));
+                }
+                if (tableHandles.isEmpty()) {
+                    return null;
+                }
+                if (tableHandles.size() > 1) {
```

Comment:
how could that happen? what does schemaTableName look like in the exception log?